### PR TITLE
[MODEL-7979] Treat single col blank values as NAN in CSVs

### PR DESCRIPTION
**Pull request by jmbannon**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
When reading CSV input files, we should treat blank lines as NaNs instead of skipping over them. This does not include the last newline.

## Changes
- set skip_blank_lines = False when only 1 col is read
- unit test
- use existing CSV reading code in shared_fit_preprocessing

### CODE REVIEW

File: custom_model_runner/datarobot_drum/drum/utils.py [hunk 1]
```diff
@@ -348,8 +343,18 @@ def read_structured_input_data_as_df(binary_data, mimetype, sparse_colnames=None
                 df.fillna(value=np.nan, inplace=True)
 
                 return df
-            else:
-                return pd.read_csv(io.BytesIO(binary_data))
+            else:  # CSV format
+                df = pd.read_csv(io.BytesIO(binary_data))
+
+                # If the DataFrame only contains a single column, treat blank lines as NANs
+                if df.shape[1] == 1:
```
Comment (jasonjmcghee) [inline custom_model_runner/datarobot_drum/drum/utils.py:350]: is it possible for the shape to be 1d?

Comment (jmbannon) [reply custom_model_runner/datarobot_drum/drum/utils.py:350]: I don't think dataframes can be 1d

Comment (jmbannon) [reply custom_model_runner/datarobot_drum/drum/utils.py:350]: (this will always return a dataframe)

Comment (jasonjmcghee) [reply custom_model_runner/datarobot_drum/drum/utils.py:350]: yup you're good: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.shape.html

**Review by jasonjmcghee — COMMENTED** [review]
You said _what_ in the PR description, but _why_ do we treat single col blank values as NaN?

**Comment by jmbannon** [comment]
> You said _what_ in the PR description, but _why_ do we treat single col blank values as NaN?

Misha and a CFDS ran into this issue using drum locally. They thought that was the expected behavior.
It makes sense since if you have two cols, `,10` would translate to `nan,10`

**Review by jasonjmcghee — APPROVED** [review]

**Review by yakov-g — APPROVED** [review]

**Pull request merged**